src/docfx/lib/markdown/ExtractTitleExtension.cs

```diff
@@ -15,40 +19,137 @@ internal static class ExtractTitleExtension
         {
             return builder.Use(document =>
             {
-                var hasVisibleNodes = false;
+                dynamic? heading = null;
```

Comment:
Consider not using dynamic as it will boot up [DLR](https://docs.microsoft.com/en-us/dotnet/framework/reflection-and-codedom/dynamic-language-runtime-overview)
```suggestion
                object? heading = null;
```